pkg/adapter/github/adapter.go

```diff
@@ -56,15 +59,20 @@ func (ra *Adapter) HandleEvent(payload interface{}, header webhooks.Header) {
 
 func (ra *Adapter) handleEvent(payload interface{}, hdr http.Header) error {
 	if ra.client == nil {
-		var err error
-		if ra.client, err = client.NewHTTPClient(
-			client.WithTarget(ra.Sink),
-			client.WithHTTPBinaryEncoding(),
-			client.WithUUIDs(),
-			client.WithTimeNow(),
-		); err != nil {
-			return err
-		}
+		ra.initClientOnce.Do(func() {
+			var err error
+			if ra.client, err = client.NewHTTPClient(
+				client.WithTarget(ra.Sink),
+				client.WithHTTPBinaryEncoding(),
+				client.WithUUIDs(),
+				client.WithTimeNow(),
+			); err != nil {
+				log.Printf("failed to create cloudevent client: %s", err)
+			}
+		})
+	}
+	if ra.client == nil {
```

Comment:
This double-check on `ra.client` is a bit strange. (I realize that the current structure requires this with `sync.Once`. Maybe extract this into a function, or call the `sync.Once` unconditionally?)

Reply:
I think calling `once.Do()` every time is the right answer. https://github.com/knative/eventing-sources/pull/240#discussion_r264792991

Reply:
That is something I'm not very comfortable to use `Sync.once`, it can not do error handling and retry well. If it fails, `ra.client` will always be `nil` without any chance to retry. Anyways, I removed the check based on your comments.

Reply:
That seems like a good reason to have an initialization method (like the `Start` method the other adapters have) so the caller can receive an error when the client is unable to start.

Reply:
In this case we need to make the initialization method public, and call it in `main.go` like below, it doesn't seem like very good. Is there any other way to do it?

```
	ra := &github.Adapter{
		Sink: *sink,
	}

        # Init cloud event client
        InitCloudEventClient()

	hook := gh.New(&gh.Config{Secret: secretToken})
	hook.RegisterEvents(ra.HandleEvent,
		gh.CommitCommentEvent,
```

Reply:
Could all this be wrapped up in a blocking `Start`? But this seems like something to open an issue about rather than blocking the PR on it.